framework/include/outputs/OutputWarehouse.h

```diff
@@ -194,6 +194,12 @@ class OutputWarehouse
   /// Reset the output system
   void reset();
 
+  /**
+   * Calls the timestepSetup function for each of the output objects
```

Comment:
`timestepSetup` is too early for solver setup where I am trying to hook up monitor, change prefix, etc. SLEPc solver does not exist yet when we call `timestepSetup`.

I moved this method from the private region to the public region because I want to call this function from `EigenProblem`.

Reply:
At the same time, I do not want to declare `EigenProblem` as friend of `OutputWarehouse`

Reply:
Adding some more doxygen for this reasoning would be appreciated, otherwise someone else like me might come along and be confused and think you made a typo and think that that method doesn't actually do what you say it does

Reply:
Will do